art/core/base_components/base_model.py

```diff
@@ -267,3 +267,7 @@ def prepare_for_metric(self: Any, data: Dict):
         preds, targets = data[PREDICTION], data[TARGET]
 
         return self.unify_type(preds), self.unify_type(targets)
+
+    @abstractmethod
```

Comment:
I thought about it for 2 days and it is pretty hard to define some "defaults". Since art main capabilities is tracking this changes of parameters between step runs I thought that forcing user to do so may be a good idea

Reply:
I see... Let's keep it this way and maybe if we test art out in some more project it'll become more obvious what choicw should we make